cfme/tests/services/test_provision_stack.py

```diff
@@ -259,6 +252,7 @@ def _cleanup_vms():
                 wait_for(lambda: provider.mgmt.delete_stack(stack_data['stack_name']),
                          delay=10, num_sec=800, message="wait for stack delete")
             stack_data['vm_name'].delete_from_provider()
+            catalog_item.orch_template.delete()
```

Comment:
Deleting template here as it needs to be cleaned up after stack . If the clean up code is inside the template method , it gets executed before stack gets cleaned up and hence fails .